sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -167,7 +167,8 @@ public virtual bool IsClosed
         private readonly string[] _sessionIds;
         private readonly EntityScopeFactory _scopeFactory;
         private readonly IList<ServiceBusPlugin> _plugins;
-        private readonly IList<ReceiverManager> _receiverManagers = new List<ReceiverManager>();
+        // deliberate usage of List instead of IList for faster enumeration and less allocations
+        private readonly List<ReceiverManager> _receiverManagers = new List<ReceiverManager>();
```

Comment:
Interesting.  I'm surprised this made a difference since `List` would be the underlying implementation for the interface....

Reply:
Did you see the benchmarks? The difference is quite a bit although in the area of nano-seconds. But the gen0 overhead is lower due to more efficient enumerator handling

Reply:
I saw, and I'm not challenging the results.   :smile:     Just saying that I found them interesting, as my intuition would be the opposite.